File: client_manage_interface/src/main/java/cn/edu/aynu/rjxy/management/service/HomeDataService.java
package cn.edu.aynu.rjxy.management.service;


import cn.edu.aynu.rjxy.pojo.TbHomeData;
import com.github.pagehelper.PageInfo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface HomeDataService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbHomeData> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbHomeData home_data);


	/**
	 * 修改
	 */
	public void update(TbHomeData home_data);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbHomeData findOne(Long homeId);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] homeIds);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbHomeData home_data, int pageNum, int pageSize);
	
}

File: cinema_upload/src/main/webapp/js/service/homeDataService.js
//服务层
app.service('homeDataService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../homeData/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../homeData/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../homeData/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../homeData/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../homeData/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../homeData/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../homeData/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}
	//根据城市id查找响应的数据
	this.findByCityId=function(cityId){
		return $http.get('../homeData/findByCityId.do?cityId='+cityId);
	}
});